### PR TITLE
Updates version and dependencies of Tensorflow to 1.8.0

### PR DESCRIPTION
**Pull request by gilbertfrancois**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,8 +28,12 @@ requirements:
     - numpy >=1.12.1
     - protobuf >=3.4.0
     - six >=1.10.0
-    - tensorboard >=1.5.0,<1.6.0
     - backports.weakref >=1.0rc1
+    - astor
+    - grpcio
+    - termcolor
+    - gast
+    - tensorboard >=1.7.0,<1.8.0
```
Comment (dhirschfeld) [inline recipe/meta.yaml]: Is this valid? `tensorflow` clearly pins `tensorboard` to `>= 1.8.0, < 1.9.0` so you'd assume there's a good reason for that?
https://github.com/tensorflow/tensorflow/blob/v1.8.0/tensorflow/tools/pip_package/setup.py#L34-L43

I'd be inclined to respect the pinning as per their `setup.py` and just update `tensorboard`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,8 +28,12 @@ requirements:
     - numpy >=1.12.1
     - protobuf >=3.4.0
     - six >=1.10.0
-    - tensorboard >=1.5.0,<1.6.0
     - backports.weakref >=1.0rc1
+    - astor
+    - grpcio
+    - termcolor
+    - gast
```
Comment (jjhelmus) [inline recipe/meta.yaml]: These new requirements should all have lower version bounds as specified in the projects setup.py specifically:
```
astor >=0.6.0
grpcio >=1.8.6
termcolor >=1.1.0
gast >=0.2.0

File: recipe/meta.yaml [hunk 3]
```diff
@@ -28,8 +28,12 @@ requirements:
     - numpy >=1.12.1
     - protobuf >=3.4.0
     - six >=1.10.0
-    - tensorboard >=1.5.0,<1.6.0
     - backports.weakref >=1.0rc1
```
Comment (jjhelmus) [inline recipe/meta.yaml]: This is only required on python 2.  A selector should be added:
```
    - backports.weakref >=1.0rc1  # [py2k]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -28,8 +28,12 @@ requirements:
     - numpy >=1.12.1
```
Comment (jjhelmus) [inline recipe/meta.yaml]: For the 1.8.0 release this should likely be updated to >=1.13.3

**Comment by dhirschfeld** [comment]
***xref:*** https://github.com/conda-forge/tensorboard-feedstock/pull/7

**Comment by gilbertfrancois** [comment]
Thank you so much for the valuable comments. They were very helpful. I've updated the requirements in line with the `setup.py` as recommended. The current automated build fails because it cannot find tensor board 1.8.0 yet. It should work as soon as the tensorboard PR is accepted.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -23,13 +23,17 @@ requirements:
...
     - six >=1.10.0
-    - tensorboard >=1.5.0,<1.6.0
-    - backports.weakref >=1.0rc1
+    - protobuf >=3.4.0
+    - tensorboard >=1.8.0, <1.9.0
+    - termcolor >=1.1.0
+    - grpcio >=1.8.6
+    - mock >=2.0.0                # [py27]
+    - enum34 >=1.1.6              # [py<34]
+    - backports.weakref >=1.0rc1  # [py<34]
```
Comment (dhirschfeld) [inline recipe/meta.yaml]: Should be `- backports.weakref  # [py2k]` I think

File: recipe/meta.yaml [hunk 6]
```diff
@@ -23,13 +23,17 @@ requirements:
...
+    - gast >=0.2.0
+    - numpy >=1.13.3
     - six >=1.10.0
-    - tensorboard >=1.5.0,<1.6.0
-    - backports.weakref >=1.0rc1
+    - protobuf >=3.4.0
+    - tensorboard >=1.8.0, <1.9.0
+    - termcolor >=1.1.0
+    - grpcio >=1.8.6
+    - mock >=2.0.0                # [py27]
```
Comment (dhirschfeld) [inline recipe/meta.yaml]: IIUC `mock` is only required for testing so can be removed from the run requirements.

File: recipe/meta.yaml (hunk 5 above)
Comment (gilbertfrancois) [reply recipe/meta.yaml]: In the setup.py it is coded as:
```
# weakref.finalize and enum were introduced in Python 3.4
if sys.version_info < (3, 4):
  REQUIRED_PACKAGES.append('backports.weakref >= 1.0rc1')
  REQUIRED_PACKAGES.append('enum34 >= 1.1.6')
```
It is still better to use `# [py2k]`?

File: recipe/meta.yaml (hunk 6 above)
Comment (gilbertfrancois) [reply recipe/meta.yaml]: Thanks, I'll do that.

**Comment by gilbertfrancois** [comment]
I keep having issues starting Tensorboard from the cli, even after this change: [0e2fe37](https://github.com/conda-forge/tensorflow-feedstock/pull/53/commits/0e2fe37f9831685bba20ae9d784108915215a1ad). Does anyone have an idea how to solve it?

File: recipe/meta.yaml (hunk 5 above)
Comment (dougalsutherland) [reply recipe/meta.yaml]: They have exactly the same effect on conda-forge, doesn't matter.

**Comment by gilbertfrancois** [comment]
The build for macOS python 2.7 gave an error. Do I have to do `conda smithy rerender` again? Or does this error have another cause?

```The command "upload_or_check_non_existence ./recipe conda-forge --channel=main -m ./.ci_support/${CONFIG}.yaml" exited with 0.```

**Comment by gilbertfrancois** [comment]
I removed `mock` from the `requirements: run:` section but forgot to put it back in the `test: requires:` section for testing with py27.  (Thank you @dhirschfeld for the hint!). Hopefully all builds will be green now :)

**Comment by dhirschfeld** [comment]
Maybe intermittent. I'll close & reopen to trigger the CI again...

**Comment by dhirschfeld** [comment]
LGTM & tests pass so happy to merge this. I'll hold off for now to give others time to review/comment...

File: recipe/meta.yaml [hunk 7]
```diff
@@ -14,7 +14,16 @@ build:
   skip: True  # [py26]
   skip: True  # [py33]
   entry_points:
-    - tensorboard = tensorflow.tensorboard.tensorboard:main
+    - freeze_graph = tensorflow.python.tools.freeze_graph:run_main
+    - toco_from_protos = tensorflow.contrib.lite.toco.python.toco_from_protos:main
+    - toco = tensorflow.contrib.lite.toco.python.toco_wrapper:main
+    - saved_model_cli = tensorflow.python.tools.saved_model_cli:main
+    # We need to keep the TensorBoard command, even though the console script
+    # is now declared by the tensorboard pip package. If we remove the
```
Comment (sdvillal) [inline recipe/meta.yaml]: Isn't conda now the one responsible for creating correctly the entrypoints from now on?

**Comment by sdvillal** [comment]
I would add entrypoint tests, as commented in https://github.com/conda-forge/tensorflow-feedstock/pull/54#issuecomment-387942011.

Also I would squash to a single commit before merging.

**Comment by gilbertfrancois** [comment]
Added tests for the cli applications, as suggested by @sdvillal.

**Comment by dhirschfeld** [comment]
The error seems a bit concerning:
```
Executing transaction: ...working... done
+ tensorboard --help
+ grep USAGE
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "/home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /lib64/libc.so.6: version `GLIBC_2.16' not found (required by /home/conda/feedstock_root/build_artifacts/tensorflow_1525944978319/_test_env_placeho/lib/python3.5/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so)
```

File: recipe/meta.yaml [hunk 8]
```diff
@@ -50,6 +50,11 @@ test:
     # Skip the import test on Linux as wheel file require a more recent
     # version of GLIBC++ than the VM used to build and test package.
     - tensorflow  # [not linux]
+  commands:
+    - tensorboard --help |grep USAGE         # [not win]
+    - freeze_graph --help |grep usage        # [not win]
+    - toco_from_protos --help |grep usage    # [not win]
+    - saved_model_cli --help |grep usage     # [not win]
```
Comment (dhirschfeld) [inline recipe/meta.yaml]: I'm not sure why `| grep usage` is required. That's explicitly testing for something we don't care about (the word usage in the help text) and seems redundant.

If the entrypoint isn't there the command will fail and hence the test will fail. If the entrypoint is there but the word *usage* isn't in the text, the test will also fail which isn't something we want. Maybe that would never happen in practice but it's still a redundant test and is preventing it being tested on Windows.

I'd be inclined to remove the `|grep usage` and test on all supported platforms.

File: recipe/meta.yaml (hunk 7 above)
Comment (dhirschfeld) [reply recipe/meta.yaml]: I think this comment isn't particularly valid for the conda packages so can be removed. Not a blocker though.

**Comment by gilbertfrancois** [comment]
Noted. I change it later today.

**Comment by dhirschfeld** [comment]
#11 seems related

**Comment by sdvillal** [comment]
End of road for entrypoint testing while we keep just repackaging the tensorflow wheels. From 1.5 tensorflow wheels are built against newer glibc - so they cannot work in our centos image. So I would revert the testing changes. Sorry to see we cannot automate testing.

As a side note, ```tensorboard --help``` alone errcodes to 1 and would make our test fail. That is why you need something like the grep pipe.

**Comment by dougalsutherland** [comment]
Yeah, that's the problem with us not building from source. :\

Would be good to run the tests on Mac and Windows, though.

**Comment by gilbertfrancois** [comment]
I did not see the above mentioned error on Linux (ubuntu 16.04.3 LTS) and macOS 10.13 when I tested on my local machines. Thank you @sdvillal for addressing the cause. I restricted now the test to macOS, since I have no possibilities to test it on Windows.

**Comment by dhirschfeld** [comment]
@gilbertfrancois - please also test on Windows. Just let the CI test for you - that's what it's there for. If it fails on Windows it's telling us something important, something we need to fix.

**Comment by gilbertfrancois** [comment]
I added the tests for Windows. 
- `tensorboard` passed with a pipe to ` findstr`
- `freeze_graph` passed
- `toco_from_protos` failed
- `saved_model_cli` unknown (did not reach this test in CI yet)

```
(C:\bld\tensorflow_1525984894103\_test_env) C:\bld\tensorflow_1525984894103\test_tmp>toco_from_protos --help 
565Traceback (most recent call last):
566  File "C:\bld\tensorflow_1525984894103\_test_env\Scripts\toco_from_protos-script.py", line 6, in <module>
567    from tensorflow.contrib.lite.toco.python.toco_from_protos import main
568ImportError: No module named 'tensorflow.contrib.lite.toco.python'
569
570(C:\bld\tensorflow_1525984894103\_test_env) C:\bld\tensorflow_1525984894103\test_tmp>IF 1 NEQ 0 exit 1 
571
```

I'm kind of stuck here, since I've no Windows machine and cannot debug. Can it be that `contrib.toco` is not included in the windows package?

**Comment by sdvillal** [comment]
It seems no toco for win, so I would just remove that test and be happy. Everything else looks great.

**Comment by gilbertfrancois** [comment]
Great! Thanks for checking.

**Comment by dhirschfeld** [comment]
I'm happy with it too and think it's ready for merging. Will do so in a few hours if no one beats me to it...

**Comment by dhirschfeld** [comment]
> *As a side note, `tensorboard --help` alone errcodes to 1 and would make our test fail. That is why you need something like the grep pipe.*

Tested and confirmed on Windows. That doesn't seem like best practice :|

**Comment by dougalsutherland** [comment]
Since `toco` / `toco_from_proto` don't work on Windows, let's just not include the entry points, right?

Also added a test of the `toco` CLI.

**Comment by gilbertfrancois** [comment]
It seems like toco is not useful as cli application at all. I saw this on macOS:
```
$ toco --help
TOCO from pip install is currently not working on command line.
Please use the python TOCO API or use
bazel run tensorflow/contrib/lite:toco -- <args> from a TensorFlow source dir.
```
I removed the entrypoint completely now. No clue why it is in the `setup.py` in the first place.

**Comment by dougalsutherland** [comment]
Merging. Thanks @gilbertfrancois / @dhirschfeld / @sdvillal / @jjhelmus!

**Comment by dhirschfeld** [comment]
Thanks @gilbertfrancois for the PR & your perseverance getting it right!

**Comment by gilbertfrancois** [comment]
Big thanks to @dhirschfeld, @dougalsutherland, @sdvillal, @jjhelmus for the great comments and guidance. It was a really nice welcome in the world of _conda-forge_. Thumbs up.

**Comment by romrozen** [comment]
Hey guy, so if I understand correctly toco is not available on Windows yet?
So if I want to transform TF model to TFLite, the options are Mac and Linux?

**Pull request merged**